src/cli/search.rs

```diff
@@ -280,41 +299,47 @@ fn print_package_info<W: Write>(package: &RepoDataRecord, mut out: W) -> io::Res
     Ok(())
 }
 
-async fn search_package_by_wildcard<W: Write>(
+async fn search_package_by_wildcard<W: Write, QF>(
     package_name: PackageName,
     package_name_filter: &str,
-    repo_data: Arc<IndexMap<(Channel, Platform), SparseRepoData>>,
+    all_package_names: Vec<String>,
+    repodata_query_func: QF,
     limit: Option<usize>,
     out: W,
-) -> miette::Result<()> {
+) -> miette::Result<()>
+where
+    QF: FnOnce(Vec<MatchSpec>) -> BoxFuture<'static, Result<Vec<RepoData>, GatewayError>>
+        + std::clone::Clone,
```

Comment:
Instead of boxing you can also make the future generic.

Reply:
done!